src/modules/simulation/gz_bridge/GZBridge.cpp

```diff
@@ -389,6 +397,36 @@ void GZBridge::poseInfoCallback(const gz::msgs::Pose_V &msg)
 
 			local_position_groundtruth.timestamp = timestamp;
 			_lpos_ground_truth_pub.publish(local_position_groundtruth);
+
+			// calculate the differential pressure
+			const float temperature_local = _temperature - LAPSE_RATE * _alt_amsl;
+			const float density_ratio = powf(_temperature / temperature_local, 4.256f);
+			const float air_density = AIR_DENSITY_MSL / density_ratio;
+			// calculate differential pressure + noise in hPa
+			const float diff_pressure_noise = generate_wgn() * 0.01f;
+			float diff_pressure;
+
+			// determine if the wind information is available
+			if (_wind_velocity.norm() > 0.001d) {
+				matrix::Vector3d Va_vec = velocity - _wind_velocity;  // calculate true airspeed by wind triangle
+				float Va = Va_vec.norm();
+				diff_pressure = matrix::sign(Va) * 0.005f * air_density  * Va * Va + diff_pressure_noise;
+
+			} else {
+				matrix::Vector3f body_velocity = matrix::Dcmf(matrix::Quatf(vehicle_attitude_groundtruth.q)).transpose() *
+								 static_cast<matrix::Vector3f>(velocity);
+				diff_pressure = matrix::sign(body_velocity(0)) * 0.005f * air_density  * body_velocity(0) * body_velocity(
+							0) + diff_pressure_noise;
+			}
+
+			// publish differential pressure
+			differential_pressure_s differential_pressure{};
+			differential_pressure.device_id = 1377548; // 1377548: DRV_DIFF_PRESS_DEVTYPE_SIM, BUS: 1, ADDR: 5, TYPE: SIMULATION
+			differential_pressure.differential_pressure_pa = (double)diff_pressure * 100.0; // hPa to Pa;
+			differential_pressure.temperature = temperature_local;
+			differential_pressure.timestamp = timestamp;
+			_differential_pressure_pub.publish(differential_pressure);
```

Comment:
this callback is getting long, maybe we put the diff pressure calculation/publishing in a function